feat(args): move args to arg_utils; add max_days_after_shipping argument

glassdimly opened the pull request:
Problem
I had 86 transaction in amazon, and 25 were unmatched. This is because the Mint transactions were recorded up to a week after the Amazon ship date.

Solution
Increase the days to consider from 3 to 7. After doing this, I had only 3 unmatched transations.

I added a new argument, `max_days_after_shipping`. However, in order to access this within `mark_best_as_matched` I would have had to pass `args` through several function calls. So, I decided to break out the args into their own file. Then, I removed `args` argument from functions within `tagger`, and obtained `args` from `arg_utils.get_args()`.

glassdimly commented on mintamazontagger/arg_utils.py:
```diff
@@ -0,0 +1,206 @@
+              'pretty common user behavior to manually change the categories. '
+              'This flag prevents tagger from wiping out that user work.'))
+    parser.add_argument(
+        '--max_days_after_shipping', type=int,
```
This arg is the only change.

jprouty commented:
Thanks for the PR! Since I only work on this project less than once a month, I figured it be more prudent to get it working for you and out sooner, than trying to go back and forth through PR review.

For learning python sake, I figured I give some coding feedback, if you're open to it:

1. By moving the args to a module level variable, you're effectively making it impossible for someone to develop a program that could run two taggers at once. In a way, by introducing `args` as a global, you're breaking encapsulation, also making it harder to unit test or possibly extend in the future.
2. Related, as this PR stands, it's odd that tagger.py is special now, re-parsing the args and storing at the module level, but then the rest of the code base uses main's parsed Args, and passes around that one instance. If that were the right change (I don't believe it to be based on (1)), you should make that change to all the files that touch args.
3. Meta-point to (2), discussing changes before making them can help reduce work by locking in on a design/style beforehand
4. General SWE advice: if you do plan to move code to a new file, it's easier to review by splitting that into multiple commits. By doing a pure move in one commit, and the new logic in another, the reviewer can clearly inspect that it's a clean move, then inspect the diff.

glassdimly commented:
Alright, thanks for the feedback. What about the substance of this PR, `--max_days_after_shipping', type=int`?

jprouty commented:
Yes sir; I piped it through in [this commit](https://github.com/jprouty/mint-amazon-tagger/commit/4ad95119ac051a63375a857a398dabaa5c518518). I didn't realize you would be active right this second, otherwise you probably could have made the changes. Sorry about that.

jprouty commented:
👍 And thanks again for the substance, this is a great option to expose for people that need to tune it!

glassdimly commented:
Hi @jprouty,

I respect your architectural principle of passing args through to each function.

However, I wanted to respond here now that I have a bit more time.

I moved the args into a separate function so that `get_args()` could be called from anywhere. This, in react-speak, would be a "[context](https://reactjs.org/docs/context.html)". Properly, `args` are global, which isn't necessarily bad if it's handled properly.

If you wanted to do functional tests with separate` args` you could do this:

```
set_args(newArgs)
my_function_to_test()
```

I moved them out like this because I don't like the passing of args through multiple functions. This, in react-speak, is called "[prop-drilling](https://codeburst.io/react-anti-pattern-prop-drilling-54474d5236bd?gi=73d6df6910a0)". This can be a problem when other arguments are added to a function, and all usages need to be refactored.

But again I think there are multiple architectures that work in different contexts!

Sorry for not making multiple commits! I had to do the things I did to get it to run rather than implementing the passing of args through multiple functions, which I wanted to avoid. I wouldn't want to commit a broken module state.

Thanks for your feedback and for incorporating these changes.
